### PR TITLE
Searching for resources added

### PR DESCRIPTION
**Pull request by barakavraham**

### CODE REVIEW

File: app/api/base.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+from flask import url_for
+import random, math
+
+subpath_api = SubpathApi(base_api, '/base', 'base')
+
+
+class SearchResourcesAPI(Resource):
+
+    def __init__(self):
+        self.reqparse = reqparse.RequestParser()
```
Comment (daniel-atbay) [inline app/api/base.py]: You don't need reqparse if you're not accepting arguments

File: app/api/base.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        }
+        for resource in add_resource:
+            current_user.army.add_item_amount(resource, add_resource[resource]['amount'])
+        current_user.army.turns -= 10
+        db.session.commit()
+
+        return add_resource
+
+    def post(self):
```
Comment (daniel-atbay) [inline app/api/base.py]: This should be `get`

File: app/api/base.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        randomPercentage = random.uniform(5.0,10.0)
+        resourceFound = math.ceil(current_user.army.field/randomPercentage)
+        return resourceFound
+
+    def can_search(self):
+        return current_user.army.turns >= 10
+
+    def add_user_resources(self):
+        add_resource = {
+            'wood': {
```
Comment (daniel-atbay) [inline app/api/base.py:26]: DYNAMIC PLSSSS

File: app/routes/base.py [hunk 4]
```diff
@@ -14,9 +14,9 @@ class UserResources:
 
 def get_user_resources():
     user_resources = {
-        "Coins": UserResources(current_user.army.coin, 'coin'),
-        "Metal": UserResources(current_user.army.metal, 'metal'),
-        "Wood": UserResources(current_user.army.wood, 'wood'),
+        "coin": UserResources(current_user.army.coin, 'coin'),
```
Comment (daniel-atbay) [inline app/routes/base.py:17]: I would do it differently.
I would have a simple list of resources, and then have a function in the user model to do a for loop on that list and return the resources like that

File: app/static/css/base.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
+.card-img-top {
+    height: 25vh;
+}
+
+.resource-img {
+    height: 35px;
+    width: 35px;
+}
```
Comment (daniel-atbay) [inline app/static/css/base.css]: new line biatch

File: app/static/js/base.js [hunk 6]
```diff
@@ -1,9 +1,43 @@
...
+                return false
+
+            $.gameApiPost('base/search_resources',{}).done(function(data){
+                setUserResources($btn, data, $userResources);
+                setCardBody($btn, data.added_resource);
+            });
+        });
+    }
+    setupResourceSearch();
+}(jQuery));
```
Comment (daniel-atbay) [inline app/static/js/base.js]: new line biatch

File: app/static/js/base.js [hunk 7]
```diff
@@ -1,9 +1,43 @@
...
-    $(this).closest('div.table-parent').find('.defense').show();
-});
\ No newline at end of file
+;(function($){
+    'use strict';
+
+    function setUserResources($btn, data, $userResources) {
+        for (let resource in data.added_resource){
+            let resource_amount = $userResources.data('army-'+resource);
+            console.log(data.added_resource[resource]['amount']);
```
Comment (daniel-atbay) [inline app/static/js/base.js]: no need for console.log

File: app/static/js/base.js [hunk 8]
```diff
@@ -1,9 +1,43 @@
...
-
-$('.defense-button').on('click', function() {
-    $(this).closest('div.table-parent').find('.attack').hide();
-    $(this).closest('div.table-parent').find('.defense').show();
-});
\ No newline at end of file
+;(function($){
+    'use strict';
+
+    function setUserResources($btn, data, $userResources) {
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
    function setUserResources($btn, { added_resource, turns }, $userResources) {
```

File: app/static/js/base.js [hunk 9]
```diff
@@ -1,9 +1,43 @@
...
+        for (let resource in data.added_resource){
+            let resource_amount = $userResources.data('army-'+resource);
+            console.log(data.added_resource[resource]['amount']);
+            $('#current-'+resource+'-amount').text(resource_amount + data.added_resource[resource]['amount']);
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
            $(`#current-${resource}-amount`).text(resource_amount + data.added_resource[resource].amount);
```

File: app/static/js/base.js [hunk 10]
```diff
@@ -1,9 +1,43 @@
...
+            $('#user-resources').data(`army-${resource}` ,resource_amount + data.added_resource[resource]['amount'])
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
            $('#user-resources').data(`army-${resource}` ,resource_amount + data.added_resource[resource].amount)
```

File: app/static/js/base.js [hunk 11]
```diff
@@ -1,9 +1,43 @@
...
+        }
+        $btn.data('turns-amount', data.turnsAmount);
+        $('#current-turns-amount').text('turns: '+data.turnsAmount);
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
        $('#current-turns-amount').text(`turns: ${turns}`);
```

File: app/api/base.py [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+        db.session.commit()
+
+        return add_resource
+
+    def post(self):
+        is_successful = self.can_search() 
+        if self.can_search():      
+            add_resource = self.add_user_resources()
+        return {
+            'turnsAmount': current_user.army.turns,
```
Comment (daniel-atbay) [inline app/api/base.py]:
```suggestion
            'turns': current_user.army.turns, 
```

File: app/static/js/base.js [hunk 13]
```diff
@@ -1,9 +1,43 @@
...
+    function setupResourceSearch() {
+        $('#search-resources-btn').on('click', function(){
+            let $btn = $(this),
+                $userResources = $('#user-resources'),
+                turnsAmount = $btn.data('turns-amount');
+
+            if (turnsAmount < 10)
+                return false
+
+            $.gameApiPost('base/search_resources',{}).done(function(data){
```
Comment (daniel-atbay) [inline app/static/js/base.js]: Change `data` to a name with more meaning

Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
            $.gameApiGet('base/search_resources').done(function(data){
```

File: app/static/js/base.js [hunk 14]
```diff
@@ -1,9 +1,43 @@
...
+
+    function setCardBody($btn, added_resource) {
+        let $card_body = $btn.closest('.card').find('.card-body');
+        $card_body.empty();
+        $card_body.append('<h5 class="card-title">Search for resources</h5>');
+        $card_body.append('<p class="card-text"> you found: </p><div class="d-flex"></div>');
+        for (let resource in added_resource) {
+            $card_body.find('div').append('<div class="col text-center mt-2"></div>');
+            $card_body.find('.col').last().append(`<p class="card-text mb-0"><img class="resource-img img-fluid" src="${added_resource[resource]['picture']}" /></p>`);
+            $card_body.find('.col').last().append(`<p class="card-text">${added_resource[resource]['amount']}</p>`);
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
            $card_body.find('.col').last().append(`<p class="card-text">${added_resource[resource].amount}</p>`);
```

File: app/static/js/base.js [hunk 15]
```diff
@@ -1,9 +1,43 @@
...
+    }
+
+    function setCardBody($btn, added_resource) {
+        let $card_body = $btn.closest('.card').find('.card-body');
+        $card_body.empty();
+        $card_body.append('<h5 class="card-title">Search for resources</h5>');
+        $card_body.append('<p class="card-text"> you found: </p><div class="d-flex"></div>');
+        for (let resource in added_resource) {
+            $card_body.find('div').append('<div class="col text-center mt-2"></div>');
+            $card_body.find('.col').last().append(`<p class="card-text mb-0"><img class="resource-img img-fluid" src="${added_resource[resource]['picture']}" /></p>`);
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
            $card_body.find('.col').last().append(`<p class="card-text mb-0"><img class="resource-img img-fluid" src="${added_resource[resource].picture}" /></p>`);
```

File: app/static/js/base.js [hunk 16]
```diff
@@ -1,9 +1,43 @@
...
+    }
+
+    function setupResourceSearch() {
+        $('#search-resources-btn').on('click', function(){
+            let $btn = $(this),
+                $userResources = $('#user-resources'),
+                turnsAmount = $btn.data('turns-amount');
+
+            if (turnsAmount < 10)
+                return false
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
                return false;
```

File: app/static/js/base.js [hunk 17]
```diff
@@ -1,9 +1,43 @@
...
+            $card_body.find('div').append('<div class="col text-center mt-2"></div>');
+            $card_body.find('.col').last().append(`<p class="card-text mb-0"><img class="resource-img img-fluid" src="${added_resource[resource]['picture']}" /></p>`);
+            $card_body.find('.col').last().append(`<p class="card-text">${added_resource[resource]['amount']}</p>`);
+        }
+    }
+
+    function setupResourceSearch() {
+        $('#search-resources-btn').on('click', function(){
+            let $btn = $(this),
+                $userResources = $('#user-resources'),
```
Comment (daniel-atbay) [inline app/static/js/base.js]: This should be a global variable

File: app/static/js/base.js [hunk 18]
```diff
@@ -1,9 +1,43 @@
...
+            if (turnsAmount < 10)
+                return false
+
+            $.gameApiPost('base/search_resources',{}).done(function(data){
+                setUserResources($btn, data, $userResources);
+                setCardBody($btn, data.added_resource);
+            });
+        });
+    }
+    setupResourceSearch();
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion

    setupResourceSearch();
    
```

File: app/api/base.py (hunk 3 above)
Comment (barakavraham) [reply app/api/base.py:26]: this is dynamic, this is the main dict, if you want to add resource for example you need to add it into that dict, i can make a new util and move it to there if you want

File: app/routes/base.py (hunk 4 above)
Comment (barakavraham) [reply app/routes/base.py:17]: this is a old list that we had, all i did is to change the key names so i will be able to use it in the html page

File: app/api/base.py (hunk 1 above)
Comment (barakavraham) [reply app/api/base.py]: oki donkey

File: app/api/base.py [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+    def post(self):
+        is_successful = self.can_search() 
+        if self.can_search():      
+            add_resource = self.add_user_resources()
+        return {
+            'turnsAmount': current_user.army.turns, 
+            'added_resource': add_resource
+            }, 200 if is_successful else 400
+
+subpath_api.add_resource(SearchResourcesAPI, '/search_resources', endpoint='search_resources')
```
Comment (daniel-atbay) [inline app/api/base.py]: NEW LINE

File: app/static/js/base.js (hunk 8 above)
Comment (barakavraham) [reply app/static/js/base.js]: i didn't know you can do that, this is great, thanks

**Review by daniel-atbay — APPROVED** [review]

**Pull request merged**